buildman/component/buildcomponent.py

```diff
@@ -668,6 +676,10 @@ def _timeout(self):
         yield From(self._set_status(ComponentStatus.TIMED_OUT))
         logger.warning("Build component with realm %s has timed out", self.builder_realm)
 
+        if self._current_job is None:
+            logger.error("Error trying to time out build. Build on realm %s has already been completed", self.builder_realm)
+            raise Return()
+
         build_id = self._current_job.repo_build.uuid
```

Comment:
Interesting that we check if `self._current_job` is not None a few lines later... looks like this build_id might need to be in that block

Reply:
Moved the build_id it in the if block